app/parse.py

```diff
@@ -15,8 +20,95 @@ class Product:
     num_of_reviews: int
 
 
-def get_all_products() -> None:
-    pass
+def setup_driver():
+    options = Options()
+    options.add_argument("--headless")
+    driver = webdriver.Chrome(options=options)
+    driver.implicitly_wait(5)
+    return driver
+
+
+def scrape_products(driver, page_url):
+    driver.get(page_url)
+    products = []
+
+    while True:
+        try:
+            load_more_button = driver.find_element(
+                By.CLASS_NAME, "btn.btn-primary.btn-block"
+            )
+            load_more_button.click()
+            time.sleep(1)
```

Comment:
Using `time.sleep(1)` is a simple way to wait for the page to load, but it can be unreliable if the network is slow or the page takes longer to load. Consider using Selenium's `WebDriverWait` to wait for specific elements to be present instead.